streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -714,14 +714,7 @@ public StreamsConfig(final Map<?, ?> props) {
     @Deprecated
     public Serde keySerde() {
         try {
-            Serde<?> serde = getConfiguredInstance(KEY_SERDE_CLASS_CONFIG, Serde.class);
-            // the default value of deprecated key serde is null
-            if (serde == null) {
-                serde = defaultKeySerde();
-            } else {
-                serde.configure(originals(), true);
-            }
-            return serde;
+            return defaultKeySerde();
         } catch (final Exception e) {
```

Comment:
I think we should improve exception handling here. We catch this in `defaultKeySerde` already... I think, it might be better to have two different `try-catchs` within `defaultKeySerde` and no try-catch here. Same for value serde.